includes/class-sensei.php

```diff
@@ -1405,6 +1407,41 @@ public function body_class( $classes ) {
 		return $classes;
 	}
 
+	/**
+	 * For course theme, add a body class with the variation.
+	 *
+	 * @param array $classes Body classes.
+	 *
+	 * @intenal
+	 *
+	 * @return array Body classes.
+	 */
+	public function maybe_add_course_theme_variation_class( $classes ) {
+
+		$is_course_theme = 'course' === wp_get_theme()->get_template();
+
+		if ( ! $is_course_theme ) {
+			return $classes;
+		}
+
+		if ( ! is_array( $classes ) ) {
+			$classes = [];
+		}
+
+		$css_string    = wp_get_global_stylesheet( [ 'variables' ] ) ?? '';
+		$property_name = '--wp--custom--course-theme-variation';
+		$pattern       = "/$property_name\s*:\s*([^;]+)/";
```

Comment:
I am not a huge fan of regex, mainly because they're so hard to grok. However, I could be pacified by the addition of a comment explaining it. 😉

Reply:
Hahah 😄 I totally get that. If you have a strong feeling for it, I can rewrite this using string manipulation too, that won't be hard at all. Just using regext to make it look cleaner. But just to check, I've added a comment here explaining the Regex https://github.com/Automattic/sensei/pull/7225/commits/a2ba4b9784df66bdb9404387114529dc4ac31810, let me know if looks good enough to you ☺️